prog22.rb

```diff
@@ -0,0 +1,31 @@
+# Q-22 Create a class 'Name' with two attributes firstname and lastname. Neither of them 
+# can be blank and the first letter of firstname must be capital. Implement using 'raise'. 
+# Try using custom Exceptions.
+# [input]
+# manoj sharma
+# [output]
+# Firstname must start with uppercase letter
+
+class Name
+
+  attr_accessor :f_name, :l_name
+
+  def initialize(fname, lname)
+    @f_name = fname
+    @l_name = lname
+    self.validate
+  end
+
+  def validate
+    raise "Firstname must start with uppercase letter" if @f_name[0] !~ /[A-Z]/ 
+    raise "Lastname cannot be blank" if @l_name.empty?
+  end
+
+  def display
+    puts "Your name is #{@f_name} #{@l_name}"
+  end
+end
+
+input_name = gets.split
+name = Name.new(input_name[0].to_s, input_name[1].to_s)
+name.display if name != nil
```

Comment:
`"akshay".instance_of? (String)` is something you can use here. Do not change for now.. this is okay. Just think of the benefits of using instance_of? instead of the current check.